addons/stock/models/stock_move.py

```diff
@@ -362,7 +362,7 @@ def _process_decrease(move, quantity):
             mls_to_unlink = set()
             # Since the move lines might have been created in a certain order to respect
             # a removal strategy, they need to be unreserved in the opposite order
-            for ml in reversed(move.move_line_ids.sorted('id')):
+            for ml in reversed(move.move_line_ids.filtered(lambda ml: not ml.picked if self.env.context.get('unreserve_unpicked_only') else True).sorted('id')):
```

Comment:
Style suggestion because it makes it easier to read (not multiple instruction on the same lines) and it avoids the `filtered`, which is a loop (so, only one loop, the `for`)
```suggestion
            for ml in reversed(move.move_line_ids.sorted('id')):
                if self.env.context.get('unreserve_unpicked_only') and ml.picked:
                    continue
```